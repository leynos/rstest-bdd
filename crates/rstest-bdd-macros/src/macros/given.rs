//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
