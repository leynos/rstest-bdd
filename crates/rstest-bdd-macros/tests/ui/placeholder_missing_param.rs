use rstest_bdd_macros::given;

#[given("a number {count:u32}")]
fn step() {}

fn main() {}
