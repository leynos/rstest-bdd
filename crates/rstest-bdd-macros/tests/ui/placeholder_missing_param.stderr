error: invalid step function signature: missing step arguments for placeholders: count. help: use `#[given] fn name(ctx: &StepContext, ...)` and valid fixtures.
 --> tests/ui/placeholder_missing_param.rs:4:4
  |
4 | fn step() {}
  |    ^^^^
